api/wasm/v0/cpp/proxy_wasm_externs.h

```diff
@@ -0,0 +1,117 @@
+/*
+ * Proxy-WASM ABI.
+ */
+// NOLINT(namespace-envoy)
+
+#pragma once
+
+#include <stddef.h>
+#include <stdint.h>
+
+//
+//  ABI calls from the host into the VM.
+//
+
+// Non-stream calls.
+
+/**
+ * Called when the VM starts for each plugin.
+ * @param root_context_id is an identifier with the lifetime of the VM which will be used for
+ * one or more related plugins and their corresponding proxy_on_configure(), proxy_on_done() and
+ * proxy_on_delete() calls. It is also be provided during creation of stream context for those
+ * plugins.
+ * @param configuration_size is the size of any configuration available via proxy_get_configuration.
+ * @return non-zero on success and zero on failure (e.g. bad configuration).
+ */
+extern "C" uint32_t proxy_on_start(uint32_t root_context_id, uint32_t configuration_size);
+/**
+ * Can be called to validate a configuration (e.g. by an xDS provider) both before proxy_on_start()
+ * to verify the VM configuration or after proxy_on_start() to verify a plugin configuation.
```

Comment:
Is this the moral equivalent of doing proto validation, but because the config can just be a string, there will be a per-plugin implementation?

Reply:
Yes, because the pluin code isn't compiled into Envoy we need to call into WASM to do the validation.